### PR TITLE
Extend Lua Scripting

### PR DESCRIPTION
**Pull request by megumumpkin**
Added new property implementation for Luna classes.
Added option to add custom parameters to pass for script launch.
Implement more bindings for `wi::scene::Scene`

### CODE REVIEW

File: Content/scripts/weather_animation.lua [hunk 1]
```diff
@@ -0,0 +1,45 @@
+runProcess(function()
```
Comment (turanszkij) [inline Content/scripts/weather_animation.lua:1]: Sample scripts should follow how the other scripts work (comments, posting start and end messages on backlog)

File: WickedEngine/wiScene_BindLua.cpp [hunk 2]
```diff
@@ -4,6 +4,13 @@
 #include "wiEmittedParticle.h"
 #include "wiTexture_BindLua.h"
 #include "wiPrimitive_BindLua.h"
+#include <LUA/lua.h>
+#include <Utility/DirectXMath.h>
```
Comment (turanszkij) [inline WickedEngine/wiScene_BindLua.cpp:8]: You must not include this, it is causing compile error on windows. On windows, we use a different version of math library. You can include wiMath.h to use math library.

**Review by turanszkij — CHANGES_REQUESTED** [review]
All new lua binding functions must be added to the documentation (no need to comment them id they are trivial). Otherwise, we will forget what was not added to documentation. It is important , because we don't have code completion/intellisense for lua.
+ the other comments I made.

File: Content/scripts/weather_animation.lua (hunk 1 above)
Comment (megumumpkin) [reply Content/scripts/weather_animation.lua:1]: For the documentation of the binding functions, where should I put this?

**Review by turanszkij — APPROVED** [review]

**Review by turanszkij — CHANGES_REQUESTED** [review]
Documentation?

**Comment by megumumpkin** [comment]
> Documentation?

Done.

**Review by turanszkij — APPROVED** [review]

**Pull request merged**